### PR TITLE
ruby: update to 3.3.5 to resolve CVE-2024-39908 and CVE-2024-49761

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade ruby to 3.3.5 to resolve CVE-2024-39908. ruby 3.3.5 contains rubygem-xml version > 3.3.2, which is not affected by CVE-2024-39908 . 

Also:
ruby: patch https://github.com/advisories/GHSA-2rxp-v6pw-ch6m
rubygem-rexml: patch https://github.com/advisories/GHSA-2rxp-v6pw-ch6m

###### Change Log  

- CVE-2024-39908
- CVE-2024-49761

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-39908

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=674310&view=results

### CODE REVIEW

**Comment by Redent0r** [comment]
Failed install test in buddy build but able to install in 3.0 container
image

**Review by jslobodzian — APPROVED** [review]

**Review by mbykhovtsev-ms — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Comment by Redent0r** [comment]
I'm not sure how it failed fast track github PR check to build, but built correctly in https://dev.azure.com/mariner-org/mariner/_build/results?buildId=674310&view=results

**Comment by Redent0r** [comment]
Fixes all rubygem build failures. Test failures are no regressions by comparing against https://dev.azure.com/mariner-org/mariner/_build?definitionId=1489

File: SPECS/rubygem-fluentd/rubygem-fluentd.spec [hunk 1]
```diff
@@ -53,12 +53,25 @@ gem install -V --local --force --install-dir %{buildroot}%{gemdir} --bindir %{bu
...
+%{_bindir}/fluent-binlog-reader.lock
+%{_bindir}/fluent-ca-generate.lock
+%{_bindir}/fluent-cap-ctl.lock
+%{_bindir}/fluent-cat.lock
+%{_bindir}/fluent-ctl.lock
+%{_bindir}/fluent-debug.lock
+%{_bindir}/fluent-gem.lock
+%{_bindir}/fluent-plugin-config-format.lock
+%{_bindir}/fluent-plugin-generate.lock
+%{_bindir}/fluentd.lock
```
Comment (jslobodzian) [inline SPECS/rubygem-fluentd/rubygem-fluentd.spec:65]: I don't think we want to package the .lock file's?  Is that what other distributions have done?

Comment (Redent0r) [reply SPECS/rubygem-fluentd/rubygem-fluentd.spec]: You are right. Updated by taking patches from https://src.fedoraproject.org/rpms/ruby/c/b7e197fb887200e4faaf8fae663a9df00bdc09d3?branch=rawhide

**Comment by mbykhovtsev-ms** [comment]
Just FY about p-tests, it appears that `rubygem-instrospect`, `rubygem-rake`, `rubygem-metaclass` have always been failing before. If they fail in the latest P-test, I will check if it is the same p-tests and then we can take the PR as it's not due to the upgrade

**Review by mbykhovtsev-ms — APPROVED** [review]
I have confirmed that p-test are failing just like before the change

**Pull request merged**